var/spack/repos/builtin/packages/py-spyder/package.py

```diff
@@ -44,12 +44,21 @@ class PySpyder(PythonPackage):
     depends_on('py-qtconsole@4.2.0:', type=('build', 'run'))
     depends_on('py-nbconvert',        type=('build', 'run'))
     depends_on('py-sphinx',           type=('build', 'run'))
-    depends_on('py-pycodestyle',      type=('build', 'run'))
+    # The pycodestyle dependency is split in two, because internally it
+    # changes its name from pep8 to pycodestyle, and spyder does not cope
+    # with this change until @3.2.0
+    # https://github.com/PyCQA/pycodestyle/issues/466
+    # https://github.com/spyder-ide/spyder/blob/master/CHANGELOG.md#version-32-2017-07-24
+    depends_on('py-pycodestyle@:1.7.1', when='@:3.1.99', type=('build', 'run'))
+    depends_on('py-pycodestyle@2.1.0:', when='@3.2.0:',  type=('build', 'run'))
     depends_on('py-pylint',           type=('build', 'run'))
     depends_on('py-psutil',           type=('build', 'run'))
     depends_on('py-qtawesome@0.4.1:', type=('build', 'run'))
     depends_on('py-qtpy@1.1.0:',      type=('build', 'run'))
-    depends_on('py-zmq',              type=('build', 'run'))
-    depends_on('py-chardet@2:',       type=('build', 'run'))
+    # not sure whether this should be here or whether it should be declared
```

Comment:
I can remove it, but we probably should discuss how to deal with this kind of "I depend on A but A must depend on B+C" kind of dependencies. Also I still plan to go through `qt` and rework the variants and at least start to go through the dependency structure...

Should this then become
```
depends_on('py-qtpy@1.1.0: ^qt+webkit'`
```
or keep the explicit `qt` dependency?

Reply:
The concretizer should be able to handle both approaches (combined or not), so from the perspective of achieving a desired result, there is no difference.

I think this is entirely a matter of opinion: a reason in favor of separating them is that the dependencies are easier to spot; a reason in favor of combining them is that if `qtpy` actually needs to be specially configured to use the webkit extension, that it would be more obvious for a user where to look if an error occurs.

Reply:
Just noting a detail - it is py-pyqt which was relevant, rather than py-qtpy. And no, I don't know what the difference is either......

As far as I can figure, py-pyqt included the Python bindings for the webkit libs if they were present in the Qt installation, and did not attempt it if they were not present.

Reply:
@scheibelp I (hopefully) clarified the comment and referred to this discussion here. I'd prefer to keep the information around, but can remove it if you prefer